queries/cow_rewards.sql

```diff
@@ -1,38 +1,67 @@
+                          SELECT tx_hash, solver
+                          FROM settlements s
+                          WHERE s.block_number = t.block_number
+                            AND s.log_index > t.log_index
+                          ORDER BY s.log_index ASC
+                          LIMIT 1
+                          ) AS settlement ON true
+                      where block_number between {{start_block}} and {{end_block}}),
+     -- Usually this would be sufficient. However some orders are not actually in the DB.
+     db_order_counts as (select solver, count(order_uid) as num_orders, tx_hash
+                         from trade_hashes
+                                  inner join orders
+                                             on order_uid = uid
+                         group by solver, tx_hash),
+     corrected_trade_counts as (select solver,
+                                       tx_hash,
+                                       num_orders,
+                                       num_trades,
+                                       num_trades - num_orders as missing_orders
+                                from db_order_counts tc
+                                         inner join lateral (
+                                    select count(order_uid) as num_trades,
+                                           tx_hash          as hash
+                                    from trade_hashes
+                                    group by tx_hash
+                                    ) as th on tc.tx_hash = th.hash),
 
--- -- V2: https://dune.com/queries/1320220
--- with
--- relevant_batch_info as (
---     select solver_address,
---            count(*) * '{{PerBatchReward}}'             as batch_reward,
---            sum(num_trades) * '{{PerTradeReward}}'      as trade_reward
---     from cow_protocol_ethereum.batches
---     join cow_protocol_ethereum.solvers
---         on solver_address = address
---     where block_time between '{{StartTime}}' and '{{EndTime}}'
---       and environment not in ('services', 'test')
---     group by solver_address
--- )
--- select
---     '0xDEf1CA1fb7FBcDC777520aa7f396b4E015F497aB'                               as token_address,
---     solver_address                                                             as receiver,
---     cast((batch_reward + trade_reward) * pow(10, 18) as decimal(38,0))::string as amount
---  from relevant_batch_info
\ No newline at end of file
+     final_counts as (select concat('0x', encode(solver, 'hex')) as receiver,
+                             count(tx_hash)                      as num_batches,
+                             sum(num_trades)                     as num_trades,
+                             sum(missing_orders)                 as missing_orders
+                      from corrected_trade_counts
+                      group by solver)
+
+select *,
+       ((50 * num_batches + 35 * num_trades) * pow(10, 18))::numeric::text as amount,
```

Comment:
Here - if we subtract `missing_orders` from above we would be excluding JIT orders. cc @fleupold

Reply:
cc also @gentrexha.

Reply:
Just for my clarity's sake: From which column would we subtract `missing_orders` in this case?